pages/calendar/components/Calendar/DailyCalendar/index.tsx

```diff
@@ -1,26 +1,55 @@
+import { useRef } from "react";
+
 import classNames from "classnames/bind";
+import dayjs, { Dayjs } from "dayjs";
+import { useShallow } from "zustand/react/shallow";
 
 import { DAYS, HOURS } from "@/constants/calendarConstants";
+import { MODAL_TYPE } from "@/constants/modalType";
+import useDragSelect from "@/hooks/useDragSelect";
 import { useDateStore } from "@/store/useDateStore";
+import { useModalStore } from "@/store/useModalStore";
 
 import styles from "./DailyCalendar.module.scss";
 
 const cn = classNames.bind(styles);
 
 export default function DailyCalendar() {
-  const { focusDate } = useDateStore();
+  const dragContainerRef = useRef<HTMLDivElement>(null);
+  const { focusDate, setScheduleStart, setScheduleEnd } = useDateStore(
+    useShallow((state) => ({
+      focusDate: state.focusDate,
+      setScheduleStart: state.setScheduleStart,
+      setScheduleEnd: state.setScheduleEnd,
+    })),
+  );
+  const { openModal } = useModalStore();
+
+  const handleDragEnd = (draggedIndex: number[]) => {
+    const startHour = draggedIndex[0];
+    const endHour = draggedIndex[draggedIndex.length - 1] + 1;
+    setScheduleStart(focusDate.hour(startHour).minute(0).second(0));
+    setScheduleEnd(focusDate.hour(endHour).minute(0).second(0));
+    openModal(MODAL_TYPE.SCHEDULE_CREATE);
+  };
+
+  const { draggedIndex } = useDragSelect(dragContainerRef, handleDragEnd);
 
   return (
     <div className={cn("container")}>
       <p className={cn("date")}>
         {focusDate.date()} <span>{DAYS[focusDate.day()]}</span>
       </p>
       <div className={cn("all-day-schedules")}></div>
-      <div className={cn("time-scroll-container")}>
-        {HOURS.map((hour) => (
+      <div className={cn("time-scroll-container")} ref={dragContainerRef}>
+        {HOURS.map((hour, i) => (
           <div className={cn("time-block")} key={hour}>
             <p className={cn("label")}>{hour.toString().padStart(2, "0")}</p>
-            <div key={hour} className={cn("schedule-block")}></div>
+            <div
+              key={hour}
+              className={cn("schedule-block", { dragged: draggedIndex.includes(i) })}
+              data-index={i}
+            ></div>
```

Comment:
```suggestion
            />
```
p4: 여기도 태그 안에 내용이 없다면 바로 `close` 해도 될 것 같아요!